### PR TITLE
Add Platform Agnostic TM Query

### PR DESCRIPTION
**Pull request by dsouzai**
In Eclipse OpenJ9, I need to add an AOT feature flag for Transactional Memory (TM) in order to ensure consistency with regard to TM across the compile and load runs. However, as it turns out, the processor query for determining whether TM is enabled is different for each platform. If it was as simple as different names, I would've just `#ifdef`'d the different queries and moved on. However, on x86, the query isn't even in the `TR::CPU` class; it's initialized as a static member of the x86 codegen class, and the `initialize` method is private that can only be accessed by the x86 codegen class. 

This PR only contains changes that provide (for the most part) a wrapper around existing TM queries. The right thing to do would be to refactor all the code into the `TR::CPU` class, but that's a rabbit hole unto itself. I figure this change should suffice for now, until someone re-architects the code.

### CODE REVIEW

**Comment by dsouzai** [comment]
@0xdaryl @Leonardo2718 could you guys please review?

File: compiler/z/env/OMRCPU.hpp [hunk 1]
```diff
@@ -122,6 +122,12 @@ class CPU : public OMR::CPU
     *     Determines whether the Transactional Memory (TM) facility is available on the current processor.
     */
    bool getSupportsTransactionalMemoryFacility();
+
+   /** \brief
+    *     Determines whether the Transactional Memory (TM) facility is available on the current processor.
+    *     Alias of getSupportsTransactionalMemoryFacility() as a platform agnostic query.
+    */
+   bool supportsTM();
```
Comment (fjeremic) [inline compiler/z/env/OMRCPU.hpp]: Not a fan of the duplication here, but until all the codegens have migrated to using the `CPU` class for CPU related queries we don't have a choice.

**Comment by dsouzai** [comment]
I'm not really sure why the compile is failing; it's failing with 

```
../../compiler/z/env/OMRCPU.cpp:213:17: error: member access into incomplete
      type 'TR::CPU'
   return self()->getSupportsTransactionalMemoryFacility();
                ^
../../compiler/env/OMRCPU.hpp:36:22: note: forward declaration of 'TR::CPU'
namespace TR { class CPU; }
                     ^
```
Why isn't `self()` returning a complete class at this point? It's a .cpp file..

**Comment by dsouzai** [comment]
Another question, if the job was
```
RUN_LINT=yes RUN_BUILD=no SPEC=linux_x86-64 PLATFORM=amd64-linux64-gcc LLVM_CONFIG=llvm-config-3.8 CLANG=clang++-3.8 CXX_PATH=clang++-3.8 CXX=clang++-3.8
```
Why is the linter build also building p and z?

**Comment by 0xdaryl** [comment]
If you're adding a new CPU query, why not give it a more descriptive name while you're at it?  e.g., `supportsTransactionalMemoryInstructions` (or something).  It can't be queried in that many places in the compiler so we can afford the readability of an expanded name.

**Comment by 0xdaryl** [comment]
Rabbit holes are what refactoring in this project are all about.  However, the more refactoring we all do the fewer of them there are.  Some aren't as deep as you think they might be.  The project appreciates any effort you (or anyone else) makes toward improving the code.

If you're not willing to take the plunge, could you open an issue in the OMR repo to track the proper refactoring effort that the eventual "someone" will need to take?

**Comment by Leonardo2718** [comment]
> I'm not really sure why the compile is failing

I believe the failure is because `compiler/z/env/OMRCPU.cpp` includes `OMRCPU.hpp` instead of `CPU.hpp`.

> Why is the linter build also building p and z?

The linter is capable of running in a "cross-linting" mode so that errors can be detected on non-x86 platforms without having to build clang on those platforms.

**Comment by dsouzai** [comment]
> I believe the failure is because compiler/z/env/OMRCPU.cpp includes OMRCPU.hpp instead of CPU.hpp.

Thanks @Leonardo2718, that seemed to do the trick.

**Comment by dsouzai** [comment]
> If you're adding a new CPU query, why not give it a more descriptive name while you're at it? e.g., supportsTransactionalMemoryInstructions

@0xdaryl I changed the query to be `supportsTransactionalMemoryInstructions` . I'll open an issue regarding the refactoring work.

**Comment by dsouzai** [comment]
@fjeremic could you re-review, considering the change to the include path.

**Comment by dsouzai** [comment]
@0xdaryl could you please re-review and kick off tests if everything looks good to you?

**Comment by 0xdaryl** [comment]
@genie-omr build all

File: compiler/env/OMRCPU.hpp [hunk 2]
```diff
@@ -131,6 +131,8 @@ class CPU
    bool isI386() { return _minorArch == TR::m_arch_i386; }
    bool isAMD64() { return _minorArch == TR::m_arch_amd64; }
```
Comment (0xdaryl) [inline compiler/env/OMRCPU.hpp]: This API should have the same Doxygen comment as the platform-specific ones.   It is the API used by architectures that don't have TM support yet (e.g., ARM32, AArch64).

Comment (dsouzai) [reply compiler/env/OMRCPU.hpp]: Good point, missed that

Comment (dsouzai) [reply compiler/env/OMRCPU.hpp]: Updated (see https://github.com/eclipse/omr/compare/9142f82165c1480c850be7146abc18c6ecc4fd46..0193bf0dfe329515a899e6ef22c35ef6fcae37ec)

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**